Bottom: 682fd858497b379d36f503b41d6d16454d86b9f9
Top:    3bf82a408628f72e895d441727cfdae4872c5366
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-29 17:22:05 +0100

Refresh of hid-lesstif-ensure-crosshair-a

---

diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 6a42706..cfb65be 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -2903,6 +2903,7 @@ static void
 lesstif_notify_crosshair_change (bool changes_complete)
 {
   static int invalidate_depth = 0;
+  Pixmap save_pixmap;
 
   if (changes_complete)
     invalidate_depth --;
@@ -2920,7 +2921,12 @@ lesstif_notify_crosshair_change (bool changes_complete)
     }
 
   if (invalidate_depth == 0 && crosshair_on)
-    DrawAttached ();
+    {
+      save_pixmap = pixmap;
+      pixmap = window;
+      DrawAttached ();
+      pixmap = save_pixmap;
+    }
 
   if (!changes_complete)
     invalidate_depth ++;
@@ -2930,6 +2936,7 @@ static void
 lesstif_notify_mark_change (bool changes_complete)
 {
   static int invalidate_depth = 0;
+  Pixmap save_pixmap;
 
   if (changes_complete)
     invalidate_depth --;
@@ -2947,7 +2954,12 @@ lesstif_notify_mark_change (bool changes_complete)
     }
 
   if (invalidate_depth == 0 && crosshair_on)
-    DrawMark ();
+    {
+      save_pixmap = pixmap;
+      pixmap = window;
+      DrawMark ();
+      pixmap = save_pixmap;
+    }
 
   if (!changes_complete)
     invalidate_depth ++;
